src/main/java/hudson/remoting/Engine.java

```diff
@@ -903,19 +903,13 @@ ExponentialRetry next(EngineListenerSplitter events) throws InterruptedException
                 events.status("Bailing out after " + DurationFormatter.format(next.timeout));
                 return null;
             } else {
-                var delaySeconds = next.delay.toSeconds();
-                events.status("Waiting " + delaySeconds + " seconds before retry");
-                TimeUnit.SECONDS.sleep(delaySeconds);
+                events.status("Waiting " + DurationFormatter.format(next.delay) + " before retry");
+                TimeUnit.SECONDS.sleep(next.delay.toSeconds());
```

Comment:
Sorry for being unclear, what I meant was actually
```suggestion
                Thread.sleep(next.delay);
```
but now I see that is Java 19+. In the meantime, for subsecond precision
```suggestion
                Thread.sleep(next.delay.toMillis());
```